apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx

```diff
@@ -0,0 +1,81 @@
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import React, {useCallback} from 'react';
+import DCDO from '@cdo/apps/dcdo';
+import SectionProgress from '../sectionProgress/SectionProgress';
+import Button from '@cdo/apps/templates/Button';
+import {setShowProgressTableV2} from '@cdo/apps/templates/currentUserRedux';
+import SectionProgressV2 from './SectionProgressV2';
+import UserPreferences from '@cdo/apps/lib/util/UserPreferences';
+
+function SectionProgressSelector({
+  showProgressTableV2,
+  setShowProgressTableV2,
+}) {
+  const onShowProgressTableV2Change = useCallback(
+    e => {
+      const shouldShowV2 = !showProgressTableV2;
+      new UserPreferences().setSortByFamilyName(shouldShowV2);
+      setShowProgressTableV2(shouldShowV2);
+    },
+    [showProgressTableV2, setShowProgressTableV2]
+  );
+  const toggleV1OrV2Button = () => (
+    <div>
+      <Button onClick={onShowProgressTableV2Change}>Toggle V1/V2</Button>
+    </div>
+  );
+
+  // If progress table is disabled, only show the v1 table.
+  if (!DCDO.get('progress-table-v2-enabled', false)) {
+    return <SectionProgress />;
+  }
+
+  // If the user has not selected manually the v1 or v2 table, show the DCDO defined default.
+  if (showProgressTableV2 === undefined) {
+    if (!DCDO.get('progress-table-v2-default-v2', false)) {
+      return (
+        <div>
+          {toggleV1OrV2Button()}
+          <SectionProgress />
+        </div>
+      );
+    }
+
+    return (
+      <div>
+        {toggleV1OrV2Button()}
+        <SectionProgressV2 />
+      </div>
+    );
+  }
+
+  // If a user has selected manually, show that version.
+  if (showProgressTableV2) {
+    return (
+      <div>
+        {toggleV1OrV2Button()}
+        <SectionProgressV2 />
+      </div>
+    );
+  }
+  return (
+    <div>
+      {toggleV1OrV2Button()}
+      <SectionProgress />
+    </div>
+  );
+}
```

Comment:
Wouldn't something like this work just as well?

```suggestion
  const v2Enabled = DCDO.get('progress-table-v2-enabled', false);
  const isPreferenceSet = showProgressTableV2 !== undefined;
  const displayV2 = v2Enabled && (isPreferenceSet ?
    showProgressTableV2
    DCDO.get('progress-table-v2-default-v2', false)
  );
  
  return (
    <div>
      {v2Enabled && toggleV1OrV2Button()}
      {displayV2 ? (<SectionProgressV2 />) : (<SectionProgress />)}
    </div>
  );
```

Reply:
I mostly went with your suggestion, however I added an early return for when v2 is completely disabled. Looks much cleaner, thanks!